libtelemetry/src/test/java/com/mapbox/android/telemetry/TelemetryClientTest.java

```diff
@@ -22,9 +28,15 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 public class TelemetryClientTest extends MockWebServerTest {
+  @Before
+  public void setup() {
+    initializeMapboxTelemetry();
```

Comment:
Similar situation as commented before.
In any case, we should avoid the inline setup and make clear what we're doing/executing on each test.
So, What about including

```java
Context mockedContext = mock(Context.class, RETURNS_DEEP_STUBS);
MapboxTelemetry.applicationContext = mockedContext;
```

or `initializeMapboxTelemetryApplicationContext()` on each test?

`MapboxTelemetry.applicationContext` is a `static` variable and as such it's good to remark that needs to be initialized on each test and setup accordingly to the subject under test making sure that its value is not shared across any other tests (**I**solated from [FIRST principles](http://agileinaflash.blogspot.com.es/2009/02/first.html)).